frameworks/Java/sabina/src/main/java/sabina/benchmark/MongoDbRepository.java

```diff
@@ -0,0 +1,106 @@
+
+import java.util.*;
+import java.util.concurrent.ThreadLocalRandom;
+
+import com.mongodb.*;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoDatabase;
+import org.bson.Document;
+
+final class MongoDbRepository implements Repository {
+    private static final String [] FORTUNES = {
+        "fortune: No such file or directory",
+        "A computer scientist is someone who fixes things that aren't broken.",
+        "After enough decimal places, nobody gives a damn.",
+        "A bad random number generator: 1, 1, 1, 1, 1, 4.33e+67, 1, 1, 1",
+        "A computer program does what you tell it to do, not what you want it to do.",
+        "Emacs is a nice operating system, but I prefer UNIX. — Tom Christaensen",
+        "Any program that runs right is obsolete.",
+        "A list is only as strong as its weakest link. — Donald Knuth",
+        "Feature: A bug with seniority.",
+        "Computers make very fast, very accurate mistakes.",
+        "<script>alert(\"This should not be displayed in a browser alert box.\");</script>",
+        "フレームワークのベンチマーク"
+    };
+
+    private MongoCollection<Document> worldCollection;
+    private MongoCollection<Document> fortuneCollection;
+
+    MongoDbRepository (Properties settings) {
+        final int PORT = parseInt (settings.getProperty ("mongodb.port"));
+        final String HOST = settings.getProperty ("mongodb.host");
+        final String DATABASE = settings.getProperty ("mongodb.database");
+        final String WORLD = settings.getProperty ("mongodb.world.collection");
+        final String FORTUNE = settings.getProperty ("mongodb.fortune.collection");
+
+        MongoClient mongoClient = new MongoClient (HOST, PORT);
+        MongoDatabase db = mongoClient.getDatabase (DATABASE);
+        worldCollection = db.getCollection (WORLD);
+        fortuneCollection = db.getCollection (FORTUNE);
+    }
+
+    @Override public List<Fortune> getFortunes () {
+        List<Fortune> fortunes = new ArrayList<> ();
+
+        fortuneCollection.find ().forEach ((Block<Document>)doc ->
+            fortunes.add (new Fortune (doc.get ("_id", Double.class).intValue (), (String)doc.get
+                ("message")))
+        );
+
+        return fortunes;
+    }
+
+    @Override public World[] getWorlds (int queries, boolean update) {
+        final World[] worlds = new World[queries];
+        final Random random = ThreadLocalRandom.current ();
+
+        for (int ii = 0; ii < queries; ii++) {
+            int id = random.nextInt (DB_ROWS) + 1;
+            worlds[ii] = update? updateWorld (id, random.nextInt (DB_ROWS) + 1) : findWorld (id);
+        }
+
+        return worlds;
+    }
+
+    private World findWorld (int id) {
+        return createWorld (worldCollection.find(eq ("_id", (double)id)).first ());
+    }
+
+    private World createWorld (Document world) {
+        try {
+            return new World (world.get ("_id", Double.class).intValue (), world.get
```

Comment:
This is ok... I create a World instance when a World is fetched from DB. But as fas as I see, creates an instance per findWorld call (which is the same as 'queries' supplied in the test).